pandas/_libs/tslibs/timedeltas.pyx

```diff
@@ -43,40 +43,227 @@ Components = collections.namedtuple('Components', [
+                                 errors=errors, name=arg.name)
+    elif is_string_object(arg):
+        pass
+    elif hasattr(arg, '__iter__') and ndim == 0:
+        # extract array scalar and process below
+        arg = arg.item()
+    elif hasattr(arg, '__iter__') and ndim == 1:
+        return _convert_listlike(arg, unit=unit, box=box, errors=errors)
+    elif ndim > 1:
+        raise TypeError('arg must be a string, timedelta, list, tuple, '
+                        '1-d array, or Series')
+
+    # ...so it must be a scalar value. Return scalar.
+    return _coerce_scalar_to_timedelta_type(arg, unit=unit,
+                                            box=box, errors=errors)
+
+
+def _validate_timedelta_unit(arg):
+    """ provide validation / translation for timedelta short units """
+    try:
+        return _unit_map[arg]
+    except:
+        if arg is None:
+            return 'ns'
+        raise ValueError("invalid timedelta unit {arg} "
+                         "provided".format(arg=arg))
+
+
+def _coerce_scalar_to_timedelta_type(r, unit='ns', box=True, errors='raise'):
+    """Convert string 'r' to a timedelta object."""
+
+    try:
+        result = convert_to_timedelta64(r, unit)
+    except ValueError:
+        if errors == 'raise':
+            raise
+        elif errors == 'ignore':
+            return r
+
+        # coerce
+        result = NaT
+
+    if box:
+        result = Timedelta(result)
+    return result
+
+
+def _convert_listlike(arg, unit='ns', box=True, errors='raise', name=None):
+    """Convert a list of objects to a timedelta index object."""
+
+    if isinstance(arg, (list, tuple)) or not hasattr(arg, 'dtype'):
+        arg = np.array(list(arg), dtype='O')
+
+    # these are shortcut-able
+    if arg.dtype.kind == 'm':
+        # Any variant of timedelta64 dtype
+        value = arg.astype('timedelta64[ns]')
+    elif arg.dtype.kind in ['i', 'u']:
+        # Any integer dtype, specifically excluding datetime64
+        in_dtype = 'timedelta64[{unit}]'.format(unit=unit)
+        value = arg.astype(in_dtype).astype('timedelta64[ns]', copy=False)
+    else:
+        from pandas._libs.algos import ensure_object
+        try:
+            value = array_to_timedelta64(ensure_object(arg),
+                                         unit=unit, errors=errors)
+            value = value.astype('timedelta64[ns]', copy=False)
+        except ValueError:
+            if errors == 'ignore':
+                return arg
+            else:
+                # This else-block accounts for the cases when errors='raise'
+                # and errors='coerce'. If errors == 'raise', these errors
+                # should be raised. If errors == 'coerce', we shouldn't
+                # expect any errors to be raised, since all parsing errors
+                # cause coercion to pd.NaT. However, if an error / bug is
+                # introduced that causes an Exception to be raised, we would
```

Comment:
i don’t think a whole lot of value of moving a top level api like this

what is the motivation?

Reply:
Yah really just unification at this point.  For me its a coin-flip as to whether its worthwhile.

Reply:
i prefer to leave self contained to cython or really perf sensitive

python is much easier to debug for most folks

so either close or strip out to minimal changes

also instead of going to new areas can work in trimming the existing list?

really appreciate all things!